l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -135,25 +139,37 @@ contract ExecutorFacet is Base, IExecutor {
             // Need to check that each log was sent by the correct address.
             if (logKey == uint256(SystemLogKey.L2_TO_L1_LOGS_TREE_ROOT_KEY)) {
                 require(logSender == L2_TO_L1_MESSENGER_SYSTEM_CONTRACT_ADDR, "lm");
-                l2LogsTreeRoot = logValue;
+                logOutput.l2LogsTreeRoot = logValue;
             } else if (logKey == uint256(SystemLogKey.TOTAL_L2_TO_L1_PUBDATA_KEY)) {
-                require(logSender == L2_TO_L1_MESSENGER_SYSTEM_CONTRACT_ADDR, "ln");
-                require(providedL2ToL1PubdataHash == logValue, "wp");
+                if (_pubdataSource == PubdataSource.Calldata) {
+                    require(logSender == L2_TO_L1_MESSENGER_SYSTEM_CONTRACT_ADDR, "ln");
+                    logOutput.pubdataHash = logValue;
+                }
             } else if (logKey == uint256(SystemLogKey.STATE_DIFF_HASH_KEY)) {
                 require(logSender == L2_TO_L1_MESSENGER_SYSTEM_CONTRACT_ADDR, "lb");
-                stateDiffHash = logValue;
+                logOutput.stateDiffHash = logValue;
             } else if (logKey == uint256(SystemLogKey.PACKED_BATCH_AND_L2_BLOCK_TIMESTAMP_KEY)) {
                 require(logSender == L2_SYSTEM_CONTEXT_SYSTEM_CONTRACT_ADDR, "sc");
-                packedBatchAndL2BlockTimestamp = uint256(logValue);
+                logOutput.packedBatchAndL2BlockTimestamp = uint256(logValue);
             } else if (logKey == uint256(SystemLogKey.PREV_BATCH_HASH_KEY)) {
                 require(logSender == L2_SYSTEM_CONTEXT_SYSTEM_CONTRACT_ADDR, "sv");
-                previousBatchHash = logValue;
+                logOutput.previousBatchHash = logValue;
             } else if (logKey == uint256(SystemLogKey.CHAINED_PRIORITY_TXN_HASH_KEY)) {
                 require(logSender == L2_BOOTLOADER_ADDRESS, "bl");
-                chainedPriorityTxsHash = logValue;
+                logOutput.chainedPriorityTxsHash = logValue;
             } else if (logKey == uint256(SystemLogKey.NUMBER_OF_LAYER_1_TXS_KEY)) {
                 require(logSender == L2_BOOTLOADER_ADDRESS, "bk");
-                numberOfLayer1Txs = uint256(logValue);
+                logOutput.numberOfLayer1Txs = uint256(logValue);
+            } else if (logKey == uint256(SystemLogKey.BLOB_ONE_HASH_KEY)) {
+                if (_pubdataSource == PubdataSource.Blob) {
```

Comment:
I don't think you should have this `if`. Since L2 contracts always send the data you can still propagate it to `logOutput` and still do "just in case" check for `logSender`. The same for `SystemLogKey.BLOB_TWO_HASH_KEY ` and `SystemLogKey.TOTAL_L2_TO_L1_PUBDATA_KEY`.

Reply:
how does this affect the batch commitment? if we are submitting pubdata via calldata then should we ignore the blob linear hashes and use the empty hash instead? @StanislavBreadless

Reply:
looking at @StanislavBreadless comment https://github.com/matter-labs/era-contracts/pull/138#discussion_r1450153684 we would want to send over 0's